pkgs/sage-conf/pyproject.toml.m4

```diff
@@ -0,0 +1,45 @@
+include(`sage_spkg_versions_toml.m4')dnl' -*- conf-toml -*-
+[build-system]
+requires = [
+    SPKG_INSTALL_REQUIRES_setuptools
```

Comment:
@mkoeppe I think this is the only m4 line in this file. What about we switch to an explicit version and make this file not preprocessed?

In fact, could we just require "setuptools" without version? It doesn't seem this pkg has big needs; a version of setuptools that supports `pyproject.toml` seems like it might be good enough.

Reply:
Three reasons:
 - this makes packages depend on sage-the-distro, as observed by @orlitzky .
 - before this PR, we have `requires = ["setuptools", "wheel"]` without version and it seems to be ok.
 - this PR is 4-3 but it's at least 5-2 (and maybe 7-0?) with this minor change.

Reply:
> * before this PR, we have `requires = ["setuptools", "wheel"]` without version and it seems to be ok.

Please, @tornaria, I just explained it, didn't I? 

This very PR is changing the metadata format. So one needs a setuptools version that supports the new format that is used.

Reply:
By the way, the very fact that you are trying to discuss the version requirements for setuptools on this PR with me illustrates why the centralization of the version requirements is a good design: 
- What versions of metadata are used in our distribution packages in `SAGE_ROOT/pkgs/` is a **cross-cutting concern**. 
- It is most efficient to make the necessary changes to it once, for all packages, rather than one by one and then repeating discussions with stakeholders in the individual packages. 

This is one of the efficiencies provided by our monorepository setup and its tooling (`./bootstrap`)

Reply:
I don't want to discuss the version requirement of setuptools at all.

If you think it's necessary, you could place a version that is good enough *for this package* (my guess is that way less than 68.1.1 might do, but that's doesn't seem so important to me as having a static `pyproject.toml` with explicit dependencies).

Pushing `setuptools` without version so CI tests on several platforms might give us a hint on what version is necessary. There's an advantage to use a lower (working) version if that's available on supported platforms.

Reply:
> * this makes packages depend on sage-the-distro

That's a polemic mischaracterization that is not helpful for a proper discussion.
- No, the packages do not "depend" on sage-the-distro.
- Only the source code of the packages in our monorepository needs an initial run of a tool (`./bootstrap`) that makes use of the centralized version metadata in `build/pkgs/`.
- Yes, these metadata and the script are also used by sage-the-distro.

Reply:
As a data point, debian stable (bookworm) is on setuptools 66.1.1. What is that missing?

Anyway, lowering the version would be an nice improvement if possible, but I'm not even asking for that. I'm asking for a static `pyproject.toml` which for this pkg is almost trivial.

Reply:
Like I wrote above:
```
> cat build/pkgs/setuptools/version_requirements.txt
> # 68.1.0 Promote pyproject.toml's [tool.setuptools] out of beta.
```
What extra info are you looking for?

Reply:
"Promote out of beta" doesn't mean it didn't work before 68.1.0, but as I said above, it's not that important.

ISTM that 61.0 might be good enough, but again, this is not so important.

Cf. https://setuptools.pypa.io/en/latest/history.html#v61-0-0

Reply:
Surely you are not suggesting that it would be an improvement to use beta-quality features for our production work.

Reply:
> Surely you are not suggesting that it would be an improvement to use beta-quality features for our production work.

I don't know if you are trying to play some kind of game here. I rather play with my kids. Here I'm trying to help you merge this PR.

Reply:
I am not; and this is an unwelcome speculation.

Reply:
Do you realize I've felt unwelcomed way before your last comment?

Reply:
BTW, care to explain why `"setuptools>=61.2"` is good enough here:
https://github.com/flatsurf/surface-dynamics/pull/82 ?

Reply:
A mistake, fixed in https://github.com/flatsurf/surface-dynamics/pull/82/commits/068a6c399f16f3fe5f58af38ae7af8e09e9f4db1; which illustrates the point that maintaining such constraints without a synchronization tool is error-prone.

Reply:
:shrug:

Reply:
> Do you realize I've felt unwelcomed way before your last comment?

Some hints, maybe helpful: 
- Pointing out that conflict could be avoided by just giving in to your demands -- I'd say that's rarely perceived as "trying to help".
- While GitHub's emoji reactions https://docs.github.com/en/rest/reactions/reactions?apiVersion=2022-11-28 are carefully curated for use in a professional environment; the full catalog of emojis is not.
- Playing offended in public ("I don't know if you are trying to play some kind of game here.") is an inappropriate manipulation, and I would strongly suggest that you refrain from such tactics in further discussions.